python/README.md

```diff
@@ -50,21 +50,39 @@ Installation
 4) Build and run the tests:
 
      $ python setup.py build
-     $ python setup.py google_test
+     $ python setup.py test
 
-     To build the C++ implementation run:
-     $ python setup.py build --cpp_implementation
+     To build, test, and use the C++ implementation, you must first compile
+     libprotobuf.so:
+
+     $ (cd .. && make)
+
+     On OS X:
```

Comment:
Nah, that caveat still applies because DYLD_LIBRARY_PATH is not even
consulted when the full path of the library is linked in. I tried every
variation I could think of, and this is the only thing that worked. If you
find another solution, I'm all ears, really.
On Apr 10, 2015 7:39 PM, "Joshua Haberman" <notifications@github.com> wrote:

> In python/README.md
> <https://github.com/google/protobuf/pull/276#discussion_r28188273>:
>
> >
> > -     To build the C++ implementation run:
> > -     $ python setup.py build --cpp_implementation
> > +     To build, test, and use the C++ implementation, you must first compile
> > +     libprotobuf.so:
> > +
> > +     $ (cd .. && make)
> > +
> > +     On OS X:
>
> I guess what I'm getting at is that it would be nice if the README didn't
> have to have an entire separate section to address OS X, brew, etc. Your
> caveat about Brew doesn't apply if the user sets DYLD_LIBRARY_PATH, right?
> Instead it could just say something like:
>
>   You must make libprotobuf.so dynamically available. You can either
>   install libprotobuf you built earlier, or set LD_LIBRARY_PATH
>   (DYLD_LIBRARY_PATH on OS X):
>
>   $ export LD_LIBRARY_PATH=../src/.libs
>   $ export DYLD_LIBRARY_PATH=../src/.libs     # for OS X
>   or
>   $ (cd .. && make install)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/google/protobuf/pull/276/files#r28188273>.
>